src/library_analyzer/processing/api/_extract_valid_values.py

```diff
@@ -0,0 +1,122 @@
+import re
+from dataclasses import dataclass, field
+
+
+@dataclass
+class Configuration:
+    _function_list: list = field(default_factory=list)
+
+    def get_function_list(self) -> list:
+        return self._function_list
+
+
+@dataclass
+class DescriptionStringConfiguration(Configuration):
+    if_listings: bool = True
+    indented_listings: bool = True
+    when_set_to: bool = True
+
+    def __post_init__(self) -> None:
+        if self.if_listings:
+            self._function_list.append(_extract_from_description_if_listings)
+        if self.indented_listings:
+            self._function_list.append(_extract_from_description_indented_listings)
+        if self.when_set_to:
+            self._function_list.append(_extract_from_descritpion_when_set_to)
+
+
+@dataclass
+class TypeStringConfiguration(Configuration):
+    curly_enum: bool = True
+    and_or_enum: bool = True
+
+    def __post_init__(self) -> None:
+        if self.curly_enum:
+            self._function_list.append(_extract_from_type_curly_enum)
+        if self.and_or_enum:
+            self._function_list.append(_extract_from_type_listing)
+
+
+def extract_valid_literals(param_description: str, param_type: str) -> set[str]:
+    # Function that extracts all valid literals from the type and description string.
```

Comment:
Use docstrings to document functions:
```suggestion
    """Function that extracts all valid literals from the type and description string."""
```